assets/src/dashboard/app/views/storyAnimTool/index.js

```diff
@@ -0,0 +1,331 @@
+import stripHTML from '../../../../edit-story/utils/stripHTML';
+import VisibleImage from '../../../../edit-story/elements/media/visibleImage';
+import ShapeLayerContent from '../../../../edit-story/elements/shape/layer';
+import {
+  SORT_DIRECTION,
+  STORY_SORT_OPTIONS,
+  STORY_STATUS,
+} from '../../../constants';
+import { PAGE_RATIO } from '../../../constants/pageStructure';
+import { PreviewPage } from '../../../components';
+import { clamp } from '../../../utils';
+import { ApiContext } from '../../api/apiProvider';
+import FontProvider from '../../font/fontProvider';
+import Timeline from './timeline';
+import {
+  PageContainer,
+  ActiveCard,
+  StorySelector,
+  Container,
+  ElementsContainer,
+  ElementInfo,
+  Type,
+  Text,
+  STORY_WIDTH,
+} from './components';
+
+function StoryAnimTool() {
+  const [activeStory, setActiveStory] = useState(null);
+  const [activeAnimation, setActiveAnimation] = useState({});
+  const [activePageIndex, setActivePageIndex] = useState(0);
+
+  const [selectedElementIds, setSelectedElementIds] = useState({});
+  const [isElementSelectable, setIsElementSelectable] = useState(false);
+
+  const {
+    actions: {
+      storyApi: { updateStory, fetchStories },
+    },
+    state: {
+      stories: { stories, storiesOrderById },
+    },
+  } = useContext(ApiContext);
+
+  useEffect(() => {
+    fetchStories({
+      searchTerm: '',
+      sortOption: STORY_SORT_OPTIONS.LAST_MODIFIED,
+      sortDirection: SORT_DIRECTION.DESC,
+      status: STORY_STATUS.ALL,
+      page: 1,
+      perPage: 25,
+    });
+  }, [fetchStories]);
+
+  const orderedStories = useMemo(() => {
+    return storiesOrderById.map((storyId) => {
+      return stories[storyId];
+    });
+  }, [stories, storiesOrderById]);
+
+  const saveActiveStoryUpdates = useCallback(() => {
+    const { content, story_data } = activeStory.originalStoryData;
+
+    updateStory({
+      ...activeStory,
+      content,
+      story_data: {
+        ...story_data,
+        pages: [...activeStory.pages],
+      },
+    });
+  }, [activeStory, updateStory]);
+
+  const handleSelectStory = useCallback(
+    (e) => {
+      if (activeStory) {
+        saveActiveStoryUpdates(activeStory);
+      }
+
+      const story = orderedStories.find(
+        (s) => s.id === parseInt(e.target.value)
+      );
+      setActiveStory(story);
+
+      //Deselect animation
+      setActiveAnimation({});
+    },
+    [activeStory, saveActiveStoryUpdates, orderedStories]
+  );
+
+  const handlePageNavClick = useCallback(
+    (offset) => {
+      const currentPageIndex = clamp(activePageIndex + offset, [
+        0,
+        activeStory.pages.length,
+      ]);
+
+      setActivePageIndex(currentPageIndex);
+    },
+    [activeStory, activePageIndex]
+  );
+
+  const handleSelectElement = useCallback(
+    (element) => {
+      if (selectedElementIds[element.id]) {
+        delete selectedElementIds[element.id];
```

Comment:
Doesn't this mutate the state in place bypassing the setter? Would it be better to create an object without this key?

Reply:
Yeah it mutates the object, but then I spread and set the object below in the `setSelectedElementIds` so it'll properly trigger a re-render with a new object, so I think it's safe to play around this way, buut I'll update so it doesn't mutate at all because that is definitely better practice :)

Reply:
Okay, I pushed up an update 👍